src/python/pants/backend/scala/compile/scalac_plugins.py

```diff
@@ -31,6 +42,53 @@ class _LoadedGlobalScalacPlugins:
     artifact_address_inputs: tuple[str, ...]
 
 
+@dataclass(frozen=True)
+class ScalaPluginsForTargetRequest:
+    target: Target
+
+
+@dataclass(frozen=True)
+class ScalaPluginTargetsForTarget:
+    plugins: Targets
+    artifacts: Targets
+
+
+@dataclass(frozen=True)
+class ScalaPluginsRequest:
+    plugins: Targets
+    artifacts: Targets
+    resolve: CoursierResolveKey
+
+    @classmethod
+    def from_target_plugins(
+        cls, seq: Iterable[ScalaPluginTargetsForTarget], resolve: CoursierResolveKey
+    ) -> ScalaPluginsRequest:
+        plugins: set[Target] = set()
+        artifacts: set[Target] = set()
+        for spft in seq:
+            plugins = plugins.union(spft.plugins)
```

Comment:
Would `plugins.update(spft.plugins)` be more idiomatic here (since this is already using a mutable `set`)?